Google login

KaviHarjani opened the pull request:
The Google login inside the webview is broken as currently uses a proxy, this should provide a better user experience, as the user is not asked to input any details and connects through google with saved accounts

KaviHarjani committed:
Google login

KaviHarjani committed:
Removing checkint, making googlesignin final

manujagobind commented on lib/webview.dart:
```diff
@@ -19,9 +20,11 @@ class _ScreenerAppState extends State<ScreenerApp> {
   late final String _screenerHomeUrl =
       widget.debug ? "http://10.0.2.2:8000" : 'https://www.screener.in';
   late String paymentUrl = "'$_screenerHomeUrl/payment/capture/'";
+  late String googleUrl = "'$_screenerHomeUrl/login/flutter_auth/'";
```
Change variable name to googleLoginUrl

manujagobind commented on lib/webview.dart:
```diff
@@ -126,7 +152,11 @@ class _ScreenerAppState extends State<ScreenerApp> {
               zoomEnabled: false,
               navigationDelegate: (NavigationRequest request) {
-                if (request.url.contains("premium")) {
+                if (request.url.endsWith('login/google/')) {
```
Use double quotes with strings for consistency.

manujagobind commented on lib/webview.dart:
```diff
@@ -62,6 +65,29 @@ class _ScreenerAppState extends State<ScreenerApp> {
+        postFunction(googleUrl, jsonEncode(_googleLogin), requestMethod);
+      }
+    } catch (error) {
+      print(error);
```
Let's not use print in production.

manujagobind commented on lib/webview.dart:
```diff
@@ -19,9 +20,11 @@ class _ScreenerAppState extends State<ScreenerApp> {
   late final String _screenerHomeUrl =
       widget.debug ? "http://10.0.2.2:8000" : 'https://www.screener.in';
   late String paymentUrl = "'$_screenerHomeUrl/payment/capture/'";
```
Are both double and single quotes needed in this string?

KaviHarjani replied:
Yes

manujagobind reviewed (commented):
Will this require a local screener back-end with the flutter branch to test?

pratyushmittal committed:
Update webview.dart

Send the access token instead of obtained data. The access token will allow the backend server to fetch the validated data on its own

KaviHarjani committed:
Passing google access token instead of the login cred, so users can't manipulate using request.POST

KaviHarjani committed:
Merge branch 'GoogleLogin-clean' of github.com:Mittal-Analytics/flutter-screener into GoogleLogin-clean

KaviHarjani committed:
Minor changes as recommended by Gobind

KaviHarjani committed:
googleLoginUrl

manujagobind commented on lib/webview.dart:
```diff
@@ -19,9 +20,11 @@ class _ScreenerAppState extends State<ScreenerApp> {
   late final String _screenerHomeUrl =
       widget.debug ? "http://10.0.2.2:8000" : 'https://www.screener.in';
```
https://www.screener.in must be in double quotes for consistency.

manujagobind commented on lib/webview.dart:
```diff
@@ -62,6 +65,26 @@ class _ScreenerAppState extends State<ScreenerApp> {
+        postFunction(googleLoginUrl, jsonEncode({'access_token': accessToken}),
+            requestMethod);
+      }
+    } catch (error) {
```
Handle the exception.

KaviHarjani replied:
What should we do in this case? show an alert box? or just pass the post request and let the function give the error?

manujagobind replied:
Not too sure what to do here. 

We can display a "failed to login" or a generic message maybe but not sure if it's required.

KaviHarjani committed:
Error message and minor double quotation around strings

manujagobind commented:
Let's address any pending comments + add fix for 403 error and get this merged.

KaviHarjani commented:
> Let's add

Fix for 403 is pushed onto screener, it will be needed to be merged with main and this should work
It would require changing google auth API, from my key to the screeners key

manujagobind commented:
> > Let's add
> 
> Fix for 403 is pushed onto screener, it will be needed to be merged with main and this should work It would require changing google auth API, from my key to the screeners key

Where do we maintain the auth key?

KaviHarjani commented:
> > > Let's add
> > 
> > 
> > Fix for 403 is pushed onto screener, it will be needed to be merged with main and this should work It would require changing google auth API, from my key to the screeners key
> 
> Where do we maintain the auth key?

we've to make a key locally, and have the SHA copied into the google console
